### PR TITLE
Some fixes and cleanups to functions/relationships/contextualization SDKs

### PR DESCRIPTION
**Pull request by erlendvollset**
a

### CODE REVIEW

File: cognite/client/_api/functions.py [hunk 1]
```diff
@@ -466,8 +465,6 @@ def _zip_and_upload_folder(self, folder: str, name: str, external_id: Optional[s
             file_id = cast(int, file.id)
 
             return file_id
-        except Exception as e:
```
Comment (alchemistake) [inline cognite/client/_api/functions.py:469]: Any reason why this is removed?

File: tests/tests_unit/test_data_classes/test_contextualization.py [hunk 2]
```diff
@@ -2,15 +2,14 @@
...
-from cognite.client import CogniteClient
 from cognite.client.data_classes import ContextualizationJob
 from tests.utils import set_env_var
 
 
 @pytest.fixture()
-def job():
+def job(cognite_client):
     with set_env_var("COGNITE_API_KEY", "BLA"):
-        return ContextualizationJob(job_id=123, status="Queued", cognite_client=CogniteClient())
```
Comment (alchemistake) [inline tests/tests_unit/test_data_classes/test_contextualization.py:13]: Nitpick: this is not related to functions, but we can keep this in this pr

**Review by alchemistake — CHANGES_REQUESTED** [review]
I believe raise on https://github.com/cognitedata/cognite-sdk-python/pull/996/files/1ea6c4e4775a7fc6f11f878ef108f3c812187ae4#diff-264fa2713547ccdb770b2891f2b90fab2fe12ad37afb09de63d4db4904c62e13L469 is necessary and if tests are passing with this removed then tests are wrong.

**Comment by erlendvollset** [comment]
> I believe raise on [cognitedata/cognite-sdk-python/pull/996/files/1ea6c4e4775a7fc6f11f878ef108f3c812187ae4#diff-264fa2713547ccdb770b2891f2b90fab2fe12ad37afb09de63d4db4904c62e13L469](https://github.com/cognitedata/cognite-sdk-python/pull/996/files/1ea6c4e4775a7fc6f11f878ef108f3c812187ae4#diff-264fa2713547ccdb770b2891f2b90fab2fe12ad37afb09de63d4db4904c62e13L469) is necessary and if tests are passing with this removed then tests are wrong.

@alchemistake Why would that be necessary? it's just reraising immediately.

File: tests/tests_unit/test_data_classes/test_contextualization.py (hunk 2 above)
Comment (erlendvollset) [reply tests/tests_unit/test_data_classes/test_contextualization.py:13]: Ah woups. I'll move it into another PR.

File: cognite/client/_api/functions.py (hunk 1 above)
Comment (erlendvollset) [reply cognite/client/_api/functions.py:469]: Looks redundant to me, unless there is some behaviour this ensures which I am missing?

**Review by alchemistake — APPROVED** [review]
I know python bad :D re-raise is not needed.

**Comment by alchemistake** [comment]
it seems like we need to fix codeowner as well

**Review by HMEiding — APPROVED** [review]
LGTM

**Review by ponnapz — APPROVED** [review]

**Pull request merged**